File: crates/semantiq-parser/tests/fixtures/kotlin/sample.kt
package com.example
import kotlin.io.println

interface Greeter {
    fun greet(): String
}

enum class Status {
    ACTIVE,
    INACTIVE
}

class User(val name: String) {
    fun greet(): String = "Hello $name"
}

object Singleton {
    fun helper() = 1
}

fun main() {
    println("hi")
}

File: crates/semantiq-parser/tests/fixtures/rust/sample.rs
use std::collections::HashMap;

/// Doc
pub struct User {
    pub name: String,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Greetable {
    fn greet(&self) -> String;
}

impl Greetable for User {
    fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }
}

pub fn process(items: &[User]) -> usize {
    items.len()
}

pub const MAX: usize = 100;

mod helpers {
    pub fn util() {}
}
